---
title: "MMRFBiolinks:Downloading and preparing files for analysis"
date: "`r BiocStyle::doc_date()`"
author:
- name: First Author
  affiliation: First Author's Affiliation
- name: Second Author
  affiliation: Second Author's Affiliation
  email: corresponding@author.com
package: MMRFBiolinksX
resource_files:
- data/dataMM.rda
output:
  BiocStyle::html_document
abstract: |
  Description of your vignette
vignette: |
  %\VignetteIndexEntry{"3. Downloading and preparing files for analysis"}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r setup}
library(MMRFBiolinksX)
```

```{r, echo=FALSE, warning=FALSE}
# This code chunk simply makes sure that all the libraries used here are installed, it will not be shown in the report (notice echo = FALSE).
packages <- c("knitr", "dplyr", "plotly","DT","TCGAbiolinks","SummarizedExperiment")
if ( length(missing_pkgs <- setdiff(packages, rownames(installed.packages()))) > 0) {
  message("Installing missing package(s): ", paste(missing_pkgs, collapse = ", "))
  install.packages(missing_pkgs)
}
```

```{r message=FALSE, warning=FALSE, include=FALSE}
library(TCGAbiolinks)
library(SummarizedExperiment)
library(dplyr)
library(DT)
```

# Downloading and preparing data for analysis
<div class="panel panel-info">
<div class="panel-heading">Data download:</div>
<div class="panel-body">


MMRFBiolinks is an extension of [TCGAbiolinks package](https://bioconductor.org/packages/TCGAbiolinks/)
which should be consulted.

</div>
</div>



<div class="panel panel-info">
<div class="panel-heading">Data prepared: SummarizedExperiment object</div>
<div class="panel-body">


A [SummarizedExperiment object](http://www.nature.com/nmeth/journal/v12/n2/fig_tab/nmeth.3252_F2.html) 
has three main matrices that can be accessed using the [SummarizedExperiment package](http://bioconductor.org/packages/SummarizedExperiment/)): 

- Sample matrix information is accessed via `colData(data)`: stores sample information. 
- Assay matrix information is accessed via `assay(data)`: stores molecular data 
- Feature matrix information (gene information) is accessed via `rowRanges(data)`: stores metadata about the features, including their genomic ranges

</div>
</div>

<div class="panel panel-warning">
<div class="panel-heading">Summarized Experiment: annotation information</div>
<div class="panel-body">


When using the function `MMRFprepare` there is an argument called `SummarizedExperiment`
which defines the output type a Summarized Experiment (default option) or a data frame.
If you set `SummarizedExperiment` to `FALSE`, you will get the data unmodified 
just as they are in the files.



</div>
</div>

## MMRFprepare 



### `Arguments`
| Argument 	| Description	|
|-------------------------------	|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|
| query 	| A query for GDCquery function 	|
| save 	| Save result as RData object? 	|
| save.filename 	| Name of the file to be save if empty an automatic will be created 	|
| directory 	| Directory/Folder where the data was downloaded. Default: GDCdata 	|
| summarizedExperiment 	| Create a summarizedExperiment? Default TRUE (if possible) 	|
| remove.files.prepared 	| Remove the files read? Default: FALSE This argument will be considered only if save argument is set to true 	|
| add.gistic2.mut 	| If a list of genes (gene symbol) is given, columns with gistic2 results from GDAC firehose (hg19) and a column indicating if there is or not mutation in that gene (hg38) (TRUE or FALSE - use the MAF file for more information) will be added to the sample matrix in the summarized Experiment object. 	|
| mut.pipeline 	| If add.gistic2.mut is not NULL this field will be taken in consideration. Four separate variant calling pipelines are implemented for GDC data harmonization. Options: muse, varscan2, somaticsniper, MuTect2. For more information: https://gdc-docs.nci.nih.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/ 	|
| mutant_variant_classification 	| List of mutant_variant_classification that will be consider a sample mutant or not. Default: "Frame_Shift_Del", "Frame_Shift_Ins", "Missense_Mutation", "Nonsense_Mutation", "Splice_Site", "In_Frame_Del", "In_Frame_Ins", "Translation_Start_Site", "Nonstop_Mutation" 	|


## Search and download data from MMRF-Compass database 

In this example we will download gene expression data from MMRF-Commpass database and  we will show object data and metadata.


```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}

query.mm.fpkm <- GDCquery(project = "MMRF-COMMPASS",
                          data.category = "Transcriptome Profiling",
                          data.type = "Gene Expression Quantification",
                          workflow.type="HTSeq - FPKM")

GDCdownload(query.mm.fpkm)

data<-MMRFprepare(query.mm.fpkm,
                  save = TRUE ,
                  save.filename = "MMCompassFPKM.rda" ,
                  directory = "GDCdata" ,
                  summarizedExperiment = TRUE)


```


```{r message=FALSE, warning=FALSE, include=FALSE}
data(dataMM)
dataMM

```
```{r message=FALSE, warning=FALSE}
# Gene expression Matrix (Only first 100 to make render faster)
datatable(as.data.frame(dataMM[1:20,]), 
              options = list(scrollX = TRUE, keys = TRUE, pageLength = 5), 
              rownames = FALSE)


```

## Search and download data for two samples from database

```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}

query.mm.fpkm <- GDCquery(project = "MMRF-COMMPASS",
                          data.category = "Transcriptome Profiling",
                          data.type = "Gene Expression Quantification",
                          workflow.type="HTSeq - FPKM",
                          barcode = c("MMRF_2473","MMRF_2111"))

GDCdownload(query.mm.fpkm)

data<-MMRFprepare(query.mm.fpkm,
                  save = TRUE ,
                  save.filename = "MMCompassFPKM.rda" ,
                  directory = "GDCdata" ,
                  summarizedExperiment = TRUE)


```


```{r message=FALSE, warning=FALSE, include=FALSE}
data(dataMM.sub)
dataMM.sub

```
```{r message=FALSE, warning=FALSE}
# Gene expression Matrix (Only first 100 to make render faster)
datatable(as.data.frame(dataMM.sub[1:20,]), 
              options = list(scrollX = TRUE, keys = TRUE, pageLength = 5), 
              rownames = FALSE)


```








# Session info {.unnumbered}

```{r sessionInfo, echo=FALSE}
sessionInfo()
```
